src/cswinrt/strings/WinRT.cs

```diff
@@ -444,6 +446,28 @@ void _UnsubscribeFromNative()
             _token.Value = 0;
         }
     }
+
+    internal unsafe class EventSource__EventHandler<T> : EventSource<System.EventHandler<T>>
+    {
+        internal EventSource__EventHandler(IObjectReference obj,
+            delegate* unmanaged[Stdcall]<System.IntPtr, System.IntPtr, out WinRT.EventRegistrationToken, int> addHandler,
+            delegate* unmanaged[Stdcall]<System.IntPtr, WinRT.EventRegistrationToken, int> removeHandler) : base(obj, addHandler, removeHandler)
+        {
+        }
+
+        override protected System.Delegate EventInvoke
+        {
+            get
+            {
+                System.EventHandler<T> handler = (System.Object obj, T e) =>
+                {
+                    _event.Invoke(obj, e);

```

Comment:
Do we need a null check?

Reply:
note you can also do _event?.Invoke(obj, e);

Reply:
I believe that would not work where the handler needs to return something. As is the case here:
```
 internal unsafe class _EventSource_global__TestComponentCSharp_EventWithReturn : EventSource<global::TestComponentCSharp.EventWithReturn>
        {
            private global::TestComponentCSharp.EventWithReturn handler;

            internal _EventSource_global__TestComponentCSharp_EventWithReturn(IObjectReference obj,
                delegate* unmanaged[Stdcall]<System.IntPtr, System.IntPtr, out WinRT.EventRegistrationToken, int> addHandler,
                delegate* unmanaged[Stdcall]<System.IntPtr, WinRT.EventRegistrationToken, int> removeHandler) : base(obj, addHandler, removeHandler)
            {
            }

            override protected System.Delegate EventInvoke
            {
                get
                {
                    if (handler == null)
                    {
                        handler = (int arg) =>
                        {
                            return _event.Invoke(arg);
                        };
                    }
                    return handler;
                }
            }
        }
```

Since the return type won't be optional, we would not be able to return optional. We cannot even return null for example, if the return type is int. One idea would be to not create any handler in case `_event` is uninitialized (or is null) and simply return `null` instead of a dummy handler that does nothing,

Reply:
How does that work today?  And what happens when an event with no delegates added is fired?

Reply:
Today, we return the default value of the return type of the handler delegate (return 0 in this case) using Expression.Default which works for reflection. To mimic that in code gen, we can write a map that contains the default value for each type and use it to return the correct default. If you think there's a simpler way, do suggest.

Reply:
do we know the return type?  if so we can do return default(type).

Reply:
like what you're already doing in write_event_invoke_return_default